qcsubmit/workflow_components/state_enumeration.py

```diff
@@ -169,30 +204,27 @@ def apply(self, molecules: List[Molecule]) -> ComponentResult:
             This is only possible using Openeye so far, if openeye is not available this step will fail.
         """
 
-        from openforcefield.utils.toolkits import OpenEyeToolkitWrapper
+        result = self._create_result(skip_unique_check=self.skip_unique_check)
 
-        result = self._create_result()
+        has_oe = self.cache
 
         # must have openeye to use this feature
-        if OpenEyeToolkitWrapper.is_available():
+        if has_oe:
 
             for molecule in molecules:
-
                 try:
                     protomers = molecule.enumerate_protomers(max_states=self.max_states)
 
-                    for protomer in protomers:
-                        result.add_molecule(protomer)
-                    result.add_molecule(molecule)
-
-                except Exception:
-                    self.fail_molecule(molecule=molecule, component_result=result)
+                    if len(protomers) == 0:
```

Comment:
we need to be careful here as we may lose the input protomer as the toolkit removes them from the return I think.